esm_runscripts/filelists.py

```diff
@@ -258,13 +264,40 @@ def target_subfolders(config):
                                 descr
                             ] = filename.replace("*", source_filename)
                         elif "/" in filename:
-                            config[model][filetype + "_targets"][
-                                descr
-                            ] = "/".join(filename.split("/")[:-1]) + "/" + source_filename.split("/")[-1]
+                            config[model][filetype + "_targets"][descr] = (
+                                "/".join(filename.split("/")[:-1])
+                                + "/"
+                                + source_filename.split("/")[-1]
+                            )
                         else:
-                            config[model][filetype + "_targets"][
-                                descr
-                            ] = source_filename.split("/")[-1]
+                            # Get the general description of the filename
+                            gen_descr = descr.split("_glob_")[0]
+                            # Load the wild cards from source and target and split them
+                            # at the *
+                            wild_card_source_all = config[model][
+                                f"{filetype}_sources_wild_card"
+                            ]
+                            wild_card_source = wild_card_source_all[gen_descr].split(
+                                "*"
+                            )
+                            wild_card_target = filename.split("*")
+                            if len(wild_card_target) != len(wild_card_source):
```

Comment:
Why do they have to have the same length. Shouldn't we also allow
`<exp_path>/restart/oifs/DATE/srf0006.*`
and we want them to end in the next run work directory as `crazy_new_name_*` such as: `<exp_path>/run_DATE/work/crazy_new_name_*`
Or is there another reason for the limitation I might have overseen?

Reply:
Not the same length, but the same number of pieces if you slice them with "*".
For example:
- `asdasd*asdasd*mmmm` copied to `a*a*m` is allowed
- `asdasd*asdasd*mmmm` copied tp `a*a` is not allowed

Reply:
Note this line:  `wild_card_target = filename.split("*")`. `wild_card_target` is not a string but a list of strings resulting from slicing the wild card with the "*". I hope this a better clarification to your question.

Reply:
I talked to @seb-wahl about it and he is find with this now.